### PR TITLE
Fix tests to have "expected value" as the first argument passed to `assert_equal`

### PR DESCRIPTION
**Pull request by pathawks**
The expected value needs to come _before_ the actual. Otherwise, the test results are confusing.

From [the docs](https://apidock.com/ruby/Test/Unit/Assertions/assert_equal):

```ruby
assert_equal(exp, act, msg = nil)
```

### CODE REVIEW

**Comment by pathawks** [comment]
I honestly can't figure out what the purpose of this test is anyway. 🤷‍♂️

**Comment by ashmaroli** [comment]
@pathawks I've updated the test comments to better express intention. Hope its clear now.

**Comment by ashmaroli** [comment]
Failing test is unrelated to this PR. Seems like an issue with JRuby archive at `https://s3.amazonaws.com/jruby.org/downloads`

File: test/test_page_without_a_file.rb [hunk 1]
```diff
@@ -56,13 +56,17 @@ def render_and_write
           "url"       => "/properties.html",
         }
         attrs.each do |prop, value|
-          # assert the props being accessible in a Jekyll::Page instance
+          # assert a couple of attributes accessible in a regular Jekyll::Page instance
           assert_equal "All the properties.\n", regular_page["content"]
           assert_equal "properties.html", regular_page["name"]
```
Comment (pathawks) [inline test/test_page_without_a_file.rb]: @ashmaroli Any idea what the purpose of these `regular_page` tests is?

Comment (ashmaroli) [reply test/test_page_without_a_file.rb]: Ah.. these should be "*variants*" too.. I'll override your commit.. stay tuned..

Comment (ashmaroli) [reply test/test_page_without_a_file.rb]: Never mind.. Its best left as is.. :grin:

**Comment by pathawks** [comment]
I think this is all ready for review, and would specifically help with some work that I am doing.

Can we get a review from @jekyll/core?

**Comment by ashmaroli** [comment]
There probably is more tests in various other files, that needs fixing. But manually going through each is not practical, I guess..

**Comment by pathawks** [comment]
> There probably is more tests in various other files, that needs fixing.

I have only been fixing them as I encounter them. If we find more in the future, we can fix them as they arise. :+1:

**Comment by ashmaroli** [comment]
manual-merge ?

**Pull request merged**